File: src/main/java/onl/tesseract/command/BasicCommandsCommand.java
package onl.tesseract.command;

import onl.tesseract.CommandsBookFactory;
import onl.tesseract.commandBuilder.CommandContext;
import onl.tesseract.commandBuilder.annotation.Command;
import onl.tesseract.commandBuilder.annotation.CommandBody;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

@Command
public class BasicCommandsCommand extends CommandContext {
    @CommandBody
    public boolean onCommand(CommandSender sender)
    {
        if (sender instanceof Player player )
        {
            CommandsBookFactory.getInstance().giveGuideBook(player);
        }
        return true;
    }
}

File: src/main/java/onl/tesseract/command/BasicCommandsCommand.kt
package onl.tesseract.command

import onl.tesseract.CommandsBookFactory.Companion.getInstance
import onl.tesseract.commandBuilder.CommandContext
import onl.tesseract.commandBuilder.annotation.Command
import onl.tesseract.commandBuilder.annotation.CommandBody
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

@Command
class BasicCommandsCommand : CommandContext() {
    @CommandBody
    fun onCommand(sender: CommandSender?): Boolean {
        if (sender is Player) {
            getInstance().giveGuideBook(sender)
        }
        return true
    }
}
